srlife/damage.py

```diff
@@ -121,11 +123,86 @@ def calculate_element_log_reliability(self, mandel_stress, temperatures, volumes
 
     svals = material.strength(temperatures)
     mvals = material.modulus(temperatures)
-    
     kvals = svals**(-mvals)
-    
+
     return -kvals * np.sum(pstress**mvals[...,None], axis = -1) * volumes
 
+class WeibullNormalTensileAveragingModel(WeibullFailureModel):
+  """
+    Weibull normal tensile average failure model
+
+    Assigning default values for nalpha and nbeta
+  """
+
+  def __init__(self, pset, *args, **kwargs):
+    super().__init__(pset, *args, **kwargs)
+    # limits and number of segments for angles
+    self.alphai = pset.get_default("alphai",0)
+    self.alphaf = pset.get_default("alphaf",np.pi)
+    self.betai = pset.get_default("betai",0)
+    self.betaf = pset.get_default("betaf",2*np.pi)
+    self.nalpha = pset.get_default("nalpha",21)
+    self.nbeta = pset.get_default("nbeta", 31)
+
+  def calculate_avg_normal_stress(self, mandel_stress, mvals, alphai, alphaf,
+  betai, betaf, nalpha, nbeta):
+    """
+        Calculate the average normal tensile stresses given the pricipal stresses
+    """
+    # Mesh grid of the vectorized angle values
+    A,B = np.meshgrid(np.linspace(alphai,alphaf,self.nalpha),
+    np.linspace(betai,betaf,self.nbeta,endpoint=False),indexing='ij')
+
+    # Increment of angles to be used in evaluating integral
+    dalpha = (A[-1,-1] - A[0,0])/(self.nalpha - 1)
+    dbeta = (B[-1,-1] - B[0,0])/(self.nbeta - 1)
+
+    # Direction cosines
+    l = np.cos(A)
+    m = np.sin(A)*np.cos(B)
+    n = np.sin(A)*np.sin(B)
+
+    # Principal stresses
+    pstress = self.calculate_principal_stress(mandel_stress)
+
+    # Normal stress
+    sigma_n = pstress[...,0,None,None]*(l**2) + pstress[...,1,None,None]*(m**2) + \
+    pstress[...,2,None,None]*(n**2)
+
+    # Area integral
+    with np.errstate(invalid='ignore'):
```

Comment:
Hmm, I would prefer not the generate a warning instead of suppressing it, but if that's not possible I suppose this would be okay.

Reply:
I tried evaluating and working with just the real value of sigma**m, as I did in Mathematica, but it still gave the error. Which is why I chose to ignore it. However, the result doesn't change, even if I remove the ignore warning comment.

Reply:
Please add a comment explaining why you are suppressing the warning so we can remember going forward -- I suspect it's because you're doing `0**0` at times.